### PR TITLE
Add Service Account Support

### PR DESCRIPTION
**Pull request by colin-msphere**
This adds DC/OS Service Account support, which is the process of creating a JWT, submitting to a login end point, and updating the Authorization token with the response.

To Do:
- [x] Change credential selection in UI
- [x] Add link to DC/OS service account documentation in help text
- [x] Fix wording of service account help text

### CODE REVIEW

**Comment by colin-msphere** [comment]
@mesosphere-ci retest this please

**Comment by maximusfloydus** [comment]
I am really looking for this functionality, any idea when it will be merged and released?

**Comment by ssk2** [comment]
@maximusfloydus: this week!

**Comment by TheSentinel454** [comment]
Any update on when this may be released?  This looks very promising!

**Comment by colin-msphere** [comment]
After the help text is reviewed, this will be merged. Release should happen one to two days afterwards.

File: src/main/resources/com/mesosphere/velocity/marathon/MarathonRecorder/help-credentialsId.html [hunk 1]
```diff
@@ -1,7 +1,36 @@
...
-        To authenticate with HTTP basic authentication, select a credential of type username with password.
-    </p>
+    <p>There are two types of authentication supported:</p>
+    <ul>
+        <li>Basic Authentication (Username and Password)</li>
+        <li>DC/OS Service Account (String)</li>
+    </ul>
+    <h3>Basic Authentication</h3>
+    <p>To authenticate with HTTP basic authentication, select a credential of type username with password. <strong>This
+        authentication type does not support DC/OS.</strong></p>
```
Comment (ssk2) [inline src/main/resources/com/mesosphere/velocity/marathon/MarathonRecorder/help-credentialsId.html]: Hm, might need to be the other way around? `DC/OS does not support this authentication type.` 

is it worth mentioning that this is typically used for open source Marathon deployments?

**Comment by Chris-Boyle** [comment]
Any idea when this will be released?

**Comment by colin-msphere** [comment]
The goal is this week and target is Thursday.

**Comment by colin-msphere** [comment]
The release is delayed due to permission problems. We are waiting on a [pull request](https://github.com/jenkins-infra/repository-permissions-updater/pull/159) within the Jenkins org. Once that is approved, the release should move forward.

Sorry for the delay.

**Pull request merged**